Src/DependencyCollector/Shared/Implementation/FrameworkHttpProcessing.cs

```diff
@@ -1,40 +1,42 @@
-﻿namespace Microsoft.ApplicationInsights.DependencyCollector.Implementation
+﻿#if !NET40
+namespace Microsoft.ApplicationInsights.DependencyCollector.Implementation
 {
     using System;
+    using System.Collections.Generic;
     using System.Globalization;
+    using System.Net;
     using Microsoft.ApplicationInsights.DataContracts;
     using Microsoft.ApplicationInsights.DependencyCollector.Implementation.Operation;
     using Microsoft.ApplicationInsights.Extensibility;
     using Microsoft.ApplicationInsights.Extensibility.Implementation;
     using Microsoft.ApplicationInsights.Web.Implementation;
 
-    internal sealed class FrameworkHttpProcessing
+    /// <summary>
+    /// Concrete class with all processing logic to generate RDD data from the callbacks
+    /// received from framework events for HTTP. This class receives and uses two types of events, which
+    /// are EventSource from FrameworkHttpEventListener, and DiagnosticSource from
+    /// HttpDiagnosticSourceListener. The challenge is, the diagnostic source events have the WebRequest
+    /// object which we need for header injection, but the events can be fired multiple times. So you won't
+    /// know if it's the first request, or if it's the last response. The event source events fire at the right
+    /// locations and just once, but they don't have the rich information. This class coordinates both events,
+    /// store information in DependencyTelemetry properly, and fire the telemetry only on the EventSource
+    /// response receive to guarantee it's done just once and at the right time.
+    /// </summary>
+    internal sealed class FrameworkHttpProcessing : HttpProcessing
```

Comment:
>Framework [](start = 26, length = 9)

`Framework` prefix meant "collected via built-in framewrok instrumentation". As opposite to `Profiler` prefix. I'd suggest to differentiate RDD collected via EventSource and Diagnostics Source and has their prefixes separately.